Add support for sub-hourly price fetching

kipe opened the pull request:
- Adds support for upcoming 15 and 30 minute pricing using "price indices" -endpoint.
  - The sub-hour resolutions are only fetched via `elspot.Prices.fetch`, no helper functions for those. For example `Prices.fetch(areas=['FI'], resolution=15)`
  - The endpoint either uses the official prices if the resolution matches, otherwise calculates an average price.
- Changes hourly fetching to use the same endpoint as 15 and 30 minute pricing, as default for some areas will be lower -> 60 minute resolution needs to be calculated as an average.
- Keeps 60 minute resolution as default for compatibility reasons.
- Adds `UnsupportedResolution` exception which is raised if an unsupported resolution is requested.


> Price indices are normalized prices for a set resolution, for one delivery area or a group of delivery areas. If official prices exist in a certain resolution, then official prices are used. If no centrally agreed official price exists for a resolution, then an average price is calculated.
> Note that settlement is always done in the lowest trading resolution for the market and area, price indices are just an indication of what the average price would be in different resolutions.
> Source: Nordpool website

kipe committed:
Split tests to multiple files

- Split tests to multiple files for easier management and better performance in editor.
- Created a utility module for VCR configuration to streamline cassette management.

kipe committed:
Update API endpoints and test cases for price fetching

- Changed API endpoint from 'DayAheadPrices' to 'DayAheadPriceIndices' for hourly price fetching.
- Updated test cases to reflect new date parameters and expected values for May 2025.
- Adjusted handling of empty area prices to return None when no data is found.

kipe committed:
Add support for 15-minute and 30-minute price resolutions

- Introduced `UnsupportedResolution` exception for unsupported resolution requests.
- Updated `Prices` class to support fetching prices at 15-minute and 30-minute intervals.
- Enhanced `fetch` and `_fetch_json` methods to accept resolution as a parameter.
- Added unit tests for 15-minute and 30-minute price fetching scenarios.
- Created test for handling unsupported resolution requests.

kipe committed:
Bump version in preparation for merge and release.

kipe committed:
Enhance country list in Prices class with additional regions

- Updated comments for clarity, specifying "Central Western Europe" and "South East Europe".
- Added Bulgaria ("BG") and Romania ("TEL") to the list of countries in the Prices class.

kipe committed:
Update Prices class to handle "SYSTEM" area for API compatibility

- Added internal handling for "SYSTEM" to be treated as "SYS" in the Prices class.
- Updated comments for clarity regarding API endpoint behavior.
- Introduced a new test case for hourly price fetching using the "SYS" area.
- Adjusted VCR cassette to reflect the new API request and response structure.

flopp999 commented:
It is working fine. I just tested it manually. I will update the examples so you can update them as well

flopp999 committed:
updated examples

flopp999 committed:
Update example.py

flopp999 committed:
Rename example.py to example1.py

flopp999 committed:
Create example2.py

flopp999 commented:
I just noticed that you need to use end_date= otherwise you will not get any data
with .hourly you do not needed to have end_date=

kipe commented:
> I just noticed that you need to use end_date= otherwise you will not get any data
> with .hourly you do not needed to have end_date=

Good catch, seems like I forgot to handle the situation when "data_type" is None -> uses incorrect parameter name in when defining areas for the request. Will fix tomorrow.

kipe commented:
> I just noticed that you need to use end_date= otherwise you will not get any data with .hourly you do not needed to have end_date=

Aah, there's no bug, just tested at the "wrong time". The library by default tries to fetch prices for tomorrow and returns `None`, if they haven't been published yet. After prices have been published, it fetches them correctly.

I'll double check this today after the prices for tomorrow have been published.

flopp999 commented:
> > I just noticed that you need to use end_date= otherwise you will not get any data with .hourly you do not needed to have end_date=
> 
> Aah, there's no bug, just tested at the "wrong time". The library by default tries to fetch prices for tomorrow and returns `None`, if they haven't been published yet. After prices have been published, it fetches them correctly.
> 
> I'll double check this today after the prices for tomorrow have been published.

I changed in the code to use today(days=0) and it then worked fine without end_date, so everything is correct

https://github.com/kipe/nordpool/blob/15min/nordpool/elspot.py#L159

flopp999 committed:
added NO5

https://data.nordpoolgroup.com/auction/day-ahead/prices?deliveryDate=latest&currency=SEK&aggregation=DeliveryPeriod&deliveryAreas=NO5

flopp999 commented:
> It is working fine. I just tested it manually. I will update the examples so you can update them as well

@kipe this is now done, I noticed on nordpool site that NO5 exists. Find changes in https://github.com/kipe/nordpool/pull/65

kipe committed:
Move examples to their own directory

- Remove example1.py and example2.py
- Add basic.py and advanced.py to examples -directory
- Update instructions for fetching Elspot prices with improved usage documentation in README.md.

kipe commented:
Thanks for the updates!
I cleaned the examples a bit, moving them to their own directory and formatted according to black rules etc.

If nothing else comes to mind, I think I'll merge and publish this tomorrow.
